### PR TITLE
 [4.0] Correcting broken Overrides Manager 

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes
The filters are not displayed in 4.0, preventing a normal use of the Manager
This PR corrects the Overrrides Manager to get the same display as in 3.x

### Testing Instructions
Before patch. No language client filters. Filter Options also broken.
Patch then make sure that 2 files are deleted:

1. `administrator/components/com_languages/Field/Languageclient.php`
it is replaced in the patch by
`administrator/components/com_languages/Field/LanguageclientField.php`

2. `administrator/components/com_languages/layouts/joomla/searchtools/default/bar.php` has to be DELETED.
We now use the new file
`administrator/components/com_languages/layouts/joomla/searchtools/default.php`

EDIT: for those using patchtester, it will work ok if bar.php is not deleted.

### Before patch
No filters to choose from in order to create overrides.
Filter Options as well as column ordering and limit box are broken.

### After patch
Filters are back.
Screen Shot 2019-04-11 at 07 27 38

Screen Shot 2019-04-10 at 17 32 19

Screen Shot 2019-04-11 at 07 34 45

@wilsonge @alikon 

Note: this patch does not prevent, for the Ajax specialists, to improve the whole stuff.
It just corrects the functionality as it works in 3.x
See discussion https://github.com/joomla/joomla-cms/pull/22107 , speciffically posts by @coolcat-creations

### CODE REVIEW

**Comment by infograf768** [comment]
Note: npm not necessary.

File: administrator/components/com_languages/Field/LanguageclientField.php [hunk 1]
```diff
@@ -7,20 +7,23 @@
...
+namespace Joomla\Component\Languages\Administrator\Field;
 
 defined('JPATH_BASE') or die;
 
 use Joomla\CMS\Form\Field\ListField;
+use Joomla\CMS\Form\FormHelper;
 use Joomla\CMS\Language\LanguageHelper;
 use Joomla\CMS\Language\Text;
 
+FormHelper::loadFieldClass('list');
```
Comment (wilsonge) [inline administrator/components/com_languages/Field/LanguageclientField.php]: This isn't required if you're using the namespaced class. Everything is autoloaded now

Comment (infograf768) [reply administrator/components/com_languages/Field/LanguageclientField.php]: nice, correcting now

**Comment by ghost** [comment]
Try to create a new Language Override got:

Screen Shot 2019-04-11 at 11 50 13

### System information
Setting | Value
-- | --
PHP Built On | Linux lamp122.cloudaccess.net 3.10.0-962.3.2.lve1.5.24.7.el6h.x86_64 nr.1 SMP Mon Dec 17 12:02:35 EST 2018 x86_64
Database Type | mysql
Database Version | 5.7.24-cll-lve
Database Collation | utf8_general_ci
Database Connection Collation | utf8_general_ci
PHP Version | 7.1.26
Web Server | Apache
WebServer to PHP Interface | cgi-fcgi
Joomla! Version | Latest Nightly Build Update from Joomla! 4.0.0-alpha8-dev Development [ Amani ] 9-March-2019 13:41 GMT
User Agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:66.0) Gecko/20100101 Firefox/66.0

**Comment by infograf768** [comment]
The file concerned by your error has been already corrected in this patch. There is no more `JFactory` in the file:
`~/administrator/components/com_languages/View/Override/HtmlView.php`

**Comment by ghost** [comment]
So first apply Patch, then create new Override?

**Comment by infograf768** [comment]
> So first apply Patch, then create new Override?

yes, you can't otherwise

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on f59858ee169c03fe4c733ba18295788e82406a38<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24553">issues.joomla.org/tracker/joomla-cms/24553</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**